Feature/log filtering

mirakst opened the pull request:
Added log module execution filter
Added log module context id filter
Added log message manager name tooltip
Ask @AndersMariegaard if you have any questions

Commit:
Implemented execution and context id filters for log modules.

Commit:
Added manager name tooltips to log messages

Mthage20 commented on DashboardFrontend/Controller.cs:
```diff
@@ -209,7 +209,7 @@ private void ParseLogMessage(LogMessage message)
                             {
                                 Name = name,
-                                ContextId = message.ContextId,
+                                ContextId = message.ContextId + 1,
```
Hva' er meningen med +1?

Mthage20 replied:
Blev besvaret over discord

KasperHenningsen commented on DashboardFrontend/DetachedWindows/LogDetached.xaml:
```diff
@@ -94,12 +94,75 @@
+                                            <TextBlock Text="{Binding Id, StringFormat='Execution {0}'}" 
+                                                               VerticalAlignment="Center"
+                                                               Foreground="White"/>
+                                            <!--<TextBlock Text="{Binding StartTime, StringFormat=' ({0:dd/MM HH:mm})'}" 
+                                                               VerticalAlignment="Center"
+                                                               Foreground="LightGray"/>-->
```
Hvad er planen med det her?
```suggestion
```

KasperHenningsen commented on DashboardFrontend/MainWindow.xaml:
```diff
@@ -316,12 +317,76 @@
+                                                    <TextBlock Text="{Binding Id, StringFormat='Execution {0}'}" 
+                                                               VerticalAlignment="Center"
+                                                               Foreground="White"/>
+                                                    <!--<TextBlock Text="{Binding StartTime, StringFormat=' ({0:dd/MM HH:mm})'}" 
+                                                               VerticalAlignment="Center"
+                                                               Foreground="LightGray"/>-->
```
igen, skal det bruges?
```suggestion
```

AndersMariegaard commented on DashboardFrontend/DetachedWindows/LogDetached.xaml:
```diff
@@ -94,12 +94,75 @@
 
-                <Button Name="ButtonLogFilter"
+                <ToggleButton Name="ButtonLogFilter"
                                 Grid.Column="9"
```
formattering virker lidt fricked overalt i den her fil

AndersMariegaard commented on DashboardFrontend/DetachedWindows/LogDetached.xaml:
```diff
@@ -94,12 +94,75 @@
+                                        <StackPanel Orientation="Horizontal">
+                                            <TextBlock Text="{Binding Id, StringFormat='Execution {0}'}" 
+                                                               VerticalAlignment="Center"
+                                                               Foreground="White"/>
+                                            <!--<TextBlock Text="{Binding StartTime, StringFormat=' ({0:dd/MM HH:mm})'}" 
+                                                               VerticalAlignment="Center"
+                                                               Foreground="LightGray"/>-->
+                                        </StackPanel>
+                                    </DataTemplate>
+                                </ComboBox.ItemTemplate>
+                            </ComboBox>
+                            <TextBlock Foreground="{DynamicResource ForegroundColor}">Context ID filter</TextBlock>
+                            <ItemsControl Height="200" 
+                                                  Background="{DynamicResource BackgroundColor}"
+                                                  Foreground="{DynamicResource ForegroundColor}"
+                                                  ItemsSource="{Binding SelectedExecution.Managers}"
+                                                  Margin="0,5">
+                                <ItemsControl.Template>
+                                    <ControlTemplate>
+                                        <ScrollViewer VerticalScrollBarVisibility="Auto" Background="{DynamicResource BackgroundColor}">
+                                            <ItemsPresenter/>
+                                        </ScrollViewer>
+                                    </ControlTemplate>
+                                </ItemsControl.Template>
+                                <ItemsControl.ItemTemplate>
+                                    <DataTemplate>
+                                        <CheckBox Click="ContextIdCheckbox_OnToggle" 
+                                                              IsChecked="{Binding IsChecked}" 
+                                                              Content="{Binding ContextId}" 
+                                                              Foreground="{DynamicResource ForegroundColor}"/>
+                                    </DataTemplate>
+                                </ItemsControl.ItemTemplate>
+                            </ItemsControl>
+                            <StackPanel Orientation="Horizontal">
+                                <Button Content="Disable all" Margin="0,0,5,0" Click="ContextIdFilter_OnClick" Tag="false"/>
+                                <Button Content="Enable all" Click="ContextIdFilter_OnClick" Tag="true"/>
+                            </StackPanel>
+                        </StackPanel>
+                    </Border>
+                </Popup>
```
Kan vi få den her smidt i en dictionary, den er rimelig grande

mirakst replied:
pls no

Commit:
Further improvements and bugfixes

Fixed formatting in xaml

Added more space to tab

Tab height now has a converter so it scales with the control, allows the context id filter control to expand as well.

Added dictionary for some styling of filter popup and it's children.

Commit:
Changed log filter popup to be a grid for better placement

Commit:
Popup grid now imitates popup behavior (close on lost focus)

This is done by an event handler which is temporarily added to the main window

Commit:
Merged dev into feature/log-filtering to update branch to origin state
